lib/Frontend/ModuleInterfaceSupport.cpp

```diff
@@ -92,6 +92,10 @@ static void printToolVersionAndFlagsComment(raw_ostream &out,
       }
     }
   }
+  if (!M->getPackageName().empty()) {
+    StringRef pkgName = M->getPackageName().str();
+    out << " -package-name " << pkgName;
+  }
```

Comment:
We shouldn't have to print the flag with custom logic. Can you try to use `ModuleInterfaceOption` instead of `ModuleInterfaceOptionIgnorablePrivate` where this flag is defined? I think it will be enough to get the desired behavior.
```
def package_name : Separate<["-"], "package-name">,
  Flags<[FrontendOption, ModuleInterfaceOptionIgnorablePrivate]>,
  HelpText<"Name of the package the module belongs to">;
```